random.go

```diff
@@ -0,0 +1,37 @@
+package chd
+
+import (
+	rand "github.com/remerge/go-xorshift"
+)
+
+func pickRandom(slice [][]byte) (v []byte) {
+	length := len(slice)
+	from := rand.Intn(length)
+	v = slice[from]
+	if len(v) > 0 {
+		return v
+	}
+
+	for i := 0; i < length; i++ {
```

Comment:
This seems too complicated. Why not?

```
for l,r := from-1,from+1; l>=0||r<len(slice); l--,r++ {
	if r >= 0 && r < len(slice) && len(slice[r]) > 0 {
		return slice[r]
	}
	if l >= 0 && l < len(slice) && len(slice[l]) > 0 {
		return slice[r]
	}
}  
```

Reply:
2. Your approach was implemented first. *BUT* corrected behaviour always tries to pick random from `l` or `r`. This give more “randomness”.

hmmm. sec.